### PR TITLE
feat: Make a QuerierRBChunk wrapper that implement QueryChunk and QueryChunkMeta

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to #4681.

This brings back [a bunch of code from the OG `DbChunk`](https://github.com/influxdata/influxdb_iox/blob/37c7ce793cc12a7f5dc29b3bc3d575c6c500f2bf/db/src/chunk.rs) when the chunk's storage was in the read buffer.

I also added a method for the read buffer cache in the catalog cache, but it's deliberately unimplemented so that it won't get used in production yet. That's so I could call it in `ChunkAdapter::new_chunk` and start to get an idea of what data I'll have when, but I'm not calling `new_chunk` anywhere yet either. 

Is everything in the place you'd expect it to be?

### CODE REVIEW

**Comment by alamb** [comment]
I will review this tomorrow morning when I am fresher. Thanks @carols10cents

File: querier/src/cache/read_buffer.rs [hunk 1]
```diff
@@ -85,8 +86,10 @@ impl ReadBufferCache {
     }
 
     /// Get read buffer chunks by Parquet file id
```
Comment (alamb) [inline querier/src/cache/read_buffer.rs:88]: This comment appears to be out of date (it is now fetched by decoded _parquet_file)

File: querier/src/chunk/mod.rs [hunk 2]
```diff
@@ -80,6 +81,98 @@ impl ChunkMeta {
     }
 }
 
+/// Chunk representation of `read_buffer::RBChunk`s for the querier.
```
Comment (alamb) [inline querier/src/chunk/mod.rs:84]: 👍

File: querier/src/chunk/mod.rs [hunk 3]
```diff
@@ -276,6 +369,69 @@ impl ParquetChunkAdapter {
             partition_sort_key,
         ))
     }
+
+    /// Create new cached chunk.
+    ///
+    /// Returns `None` if some data required to create this chunk is already gone from the catalog.
+    pub async fn new_chunk(
```
Comment (alamb) [inline querier/src/chunk/mod.rs:376]:
```suggestion
    pub async fn new_rb_chunk(
```

Or maybe 

```suggestion
    pub async fn new_cached_chunk(
```

There are already so many "chunk"s in the system, making it a little more specific I think will help

File: querier/src/chunk/mod.rs [hunk 4]
```diff
@@ -276,6 +369,69 @@ impl ParquetChunkAdapter {
...
+            .table()
+            .name(parquet_file.table_id)
+            .await?;
+
+        let iox_metadata = &decoded_parquet_file.iox_metadata;
+
+        // Somewhat hacky workaround because of implicit chunk orders, use min sequence number and
+        // hope it doesn't overflow u32. Order is non-zero, se we need to add 1.
+        let order = ChunkOrder::new(1 + iox_metadata.min_sequence_number.get() as u32)
+            .expect("cannot be zero");
```
Comment (alamb) [inline querier/src/chunk/mod.rs:407]: I wonder if @crepererum  has any thoughts on this

File: querier/src/chunk/query_access.rs [hunk 5]
```diff
@@ -135,3 +158,328 @@ impl QueryChunk for QuerierParquetChunk {
...
+
+        let mut values = self.rb_chunk.column_values(
+            rb_predicate,
+            schema::selection::Selection::Some(&[column_name]),
+            BTreeMap::new(),
+        )?;
+
+        // The InfluxRPC frontend only supports getting column values
+        // for one column at a time (this is a restriction on the Influx
+        // Read gRPC API too). However, the Read Buffer support multiple
```
Comment (alamb) [inline querier/src/chunk/query_access.rs:286]:
```suggestion
        // Read gRPC API too). However, the Read Buffer supports multiple
```

File: querier/src/chunk/query_access.rs [hunk 6]
```diff
@@ -135,3 +158,328 @@ impl QueryChunk for QuerierParquetChunk {
...
+        let rb = self.read_results.next();
+        if let Some(rb) = &rb {
+            ctx.set_metadata("output_rows", rb.num_rows() as i64);
+        }
+
+        Poll::Ready(Ok(rb).transpose())
+    }
+
+    // TODO is there a useful size_hint to pass?
+}
```
Comment (alamb) [inline querier/src/chunk/query_access.rs:485]: The code looks good in this module, but I think we need to add test coverage. The fullest coverage is probably by adding ReadBufferChunks to the setups created in `query_tests` -- maybe we can add a ticket to track that task (I think @crepererum has some ideas)

**Review by alamb — APPROVED** [review]
The only significant concern I have is lack of test coverage, but I think it would be ok to add such coverage as a follow on PR

File: querier/src/chunk/mod.rs (hunk 4 above)
Comment (carols10cents) [reply querier/src/chunk/mod.rs:407]: This was copy-pasted from [`new_querier_parquet_chunk` above](https://github.com/influxdata/influxdb_iox/blob/16e7a6d5962f610e4b21432f568865719f494edb/querier/src/chunk/mod.rs#L249-L252), I'm not sure if it applies anymore though?

File: querier/src/chunk/query_access.rs (hunk 6 above)
Comment (carols10cents) [reply querier/src/chunk/query_access.rs:485]: Filed https://github.com/influxdata/influxdb_iox/issues/4720 and added it to the list in https://github.com/influxdata/influxdb_iox/issues/4392 !

File: querier/src/chunk/mod.rs (hunk 4 above)
Comment (crepererum) [reply querier/src/chunk/mod.rs:405]: I think it's not too hacky. We might want to extend the chunk order integer width to 64bits now that OG is gone.

**Pull request merged**